networkit/cpp/embedding/Node2Vec.cpp

```diff
@@ -31,9 +31,18 @@ namespace NetworKit {
 
 Node2Vec::Node2Vec(const Graph &G, double P, double Q, count L, count N, count D)
     : G(&G), P(P), Q(Q), L(L), N(N), D(D) {
+    if (G.isDirected()) {
+        throw std::runtime_error("Current implementation can only deal with undirected graphs");
+    }
     if (G.numberOfNodes() != G.upperNodeIdBound()) {
         throw std::runtime_error("The node ids of the graph must be continuous.");
     }
+    bool hasIsolatedNodes = false;
+    for (node u : G.nodeRange())
+        if (G.degree(u) == 0)
+            hasIsolatedNodes = true;
```

Comment:
Weird that throwing inside the loop does not work. Anyway, I'd suggest to stop the loop once we find an isolated node:
```suggestion
        if (G.degree(u) == 0) {
            hasIsolatedNodes = true;
            break;
        }
```